ci-operator/step-registry/cucushift/installer/check/gcp/confidential-computing/cucushift-installer-check-gcp-confidential-computing-commands.sh

```diff
@@ -26,36 +26,88 @@ fi
 ## Try the validation
 ret=0
 
+compute_node_cc_type=""
+compute_node_maintenance_policy=""
+if [[ -n "${COMPUTE_CONFIDENTIAL_COMPUTE}" ]] && [[ -n "${COMPUTE_ON_HOST_MAINTENANCE}" ]]; then
+  compute_node_cc_type="${COMPUTE_CONFIDENTIAL_COMPUTE}"
+  compute_node_maintenance_policy="${COMPUTE_ON_HOST_MAINTENANCE}"
+elif [[ -n "${CONFIDENTIAL_COMPUTE}" ]] && [[ -n "${ON_HOST_MAINTENANCE}" ]]; then
+  compute_node_cc_type="${CONFIDENTIAL_COMPUTE}"
+  compute_node_maintenance_policy="${ON_HOST_MAINTENANCE}"
+fi
+
+control_plane_node_cc_type=""
+control_plane_node_maintenance_policy=""
+if [[ -n "${CONTROL_PLANE_CONFIDENTIAL_COMPUTE}" ]] && [[ -n "${CONTROL_PLANE_ON_HOST_MAINTENANCE}" ]]; then
+  control_plane_node_cc_type="${CONTROL_PLANE_CONFIDENTIAL_COMPUTE}"
+  control_plane_node_maintenance_policy="${CONTROL_PLANE_ON_HOST_MAINTENANCE}"
+elif [[ -n "${CONFIDENTIAL_COMPUTE}" ]] && [[ -n "${ON_HOST_MAINTENANCE}" ]]; then
+  control_plane_node_cc_type="${CONFIDENTIAL_COMPUTE}"
+  control_plane_node_maintenance_policy="${ON_HOST_MAINTENANCE}"
+fi
+
 echo "$(date -u --rfc-3339=seconds) - Checking Confidential Computing settings of cluster machines..."
 readarray -t machines < <(gcloud compute instances list --filter="name~${CLUSTER_NAME}" --format="table(name,zone)" | grep -v NAME)
```

Comment:
Why not get the exact machines from `oc get nodes -o json | jq -r '.items[].metadata.name'`, but not worry about if some unexpected machines with similar name to `CLUSTER_NAME` to be filtered mistakenly.